src/kernels/TotalStressDivergenceNEML.C

```diff
@@ -0,0 +1,162 @@
+{}
+
+void 
+TotalStressDivergenceNEML::initialSetup() {
+  for (unsigned int i = 0; i < _ndisp; i++) {
+    _disp_nums[i] = coupled("displacements", i);
+    _disp_vars[i] = getVar("displacements", i);
+    _grad_disp[i] = &coupledGradient("displacements", i);
+  }
+}
+
+Real TotalStressDivergenceNEML::computeQpResidual() {
+  if (_ld) {
+    return largeDeformationResidual(_component, _grad_test[_i][_qp]);
+  }
+  else {
+    return smallDeformationResidual(_component, _grad_test[_i][_qp]);
+  }
+}
+
+Real
+TotalStressDivergenceNEML::largeDeformationResidual(unsigned int i,
+                                                    const RealGradient & grad_phi)
+{
+  return _detJ[_qp] * _stress[_qp].row(i) * (_inv_def_grad[_qp].transpose()
+                                             * grad_phi);
+}
+
+Real
+TotalStressDivergenceNEML::smallDeformationResidual(unsigned int i,
+                                                    const RealGradient & grad_phi)
+{
+  return _stress[_qp].row(i) * grad_phi;
+}
+
+Real TotalStressDivergenceNEML::computeQpJacobian() {
+  Real value = 0.0;
+  
+  if (_ld) {
+    value += largeDeformationMatJac(_component, _component,
+                                    _grad_test[_i][_qp],
+                                    _grad_phi[_j][_qp]);
+    value += largeDeformationGeoJac(_component, _component,
+                                    _grad_test[_i][_qp],
+                                    _grad_phi[_j][_qp]);
+  }
+  else {
+    value += smallDeformationMatJac(_component, _component,
+                                    _grad_test[_i][_qp],
+                                    _grad_phi[_j][_qp]);
+  }
+
+  return value;
+}
+
+Real TotalStressDivergenceNEML::computeQpOffDiagJacobian(unsigned int jvar) {
+  Real value = 0.0;
+
+  for (unsigned int cc = 0; cc < _ndisp; cc++) {
+    if (jvar == _disp_nums[cc]) {
+      if (_ld) {
+        value += largeDeformationMatJac(_component, cc,
+                                        _grad_test[_i][_qp],
+                                        _disp_vars[cc]->gradPhi()[_j][_qp]);
+        value += largeDeformationGeoJac(_component, cc,
+                                        _grad_test[_i][_qp],
+                                        _disp_vars[cc]->gradPhi()[_j][_qp]);
+      }
+      else {
+        value += smallDeformationMatJac(_component, cc,
+                                        _grad_test[_i][_qp],
+                                        _disp_vars[cc]->gradPhi()[_j][_qp]);
+      }
+      break;
+    }
+  }
+
+  return value;
+}
+
+Real 
+TotalStressDivergenceNEML::smallDeformationMatJac(
+    unsigned int i, unsigned int k, const RealGradient & grad_phi,
+    const RealGradient & grad_psi)
+{
+  Real value = 0.0;
+  for (unsigned int j = 0; j < _ndisp; j++) {
+    for (unsigned int l = 0; l < _ndisp; l++) {
+      value += _material_jacobian[_qp](i,j,k,l) * grad_phi(j) * grad_psi(l);
+    }
+  }
+  return value;
```

Comment:
as before remove { } for one line or nested one-line loops

Reply:
Sure